homeassistant/components/tado/climate.py

```diff
@@ -500,49 +514,64 @@ def _async_update_zone_data(self):
         self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
 
     @callback
-    def _async_update_zone_callback(self):
+    def _async_update_zone_callback(self) -> None:
         """Load tado data and update state."""
         self._async_update_zone_data()
         self.async_write_ha_state()
 
     @callback
-    def _async_update_home_data(self):
+    def _async_update_home_data(self) -> None:
         """Load tado geofencing data into zone."""
         self._tado_geofence_data = self._tado.data["geofence"]
 
     @callback
-    def _async_update_home_callback(self):
+    def _async_update_home_callback(self) -> None:
         """Load tado data and update state."""
         self._async_update_home_data()
         self.async_write_ha_state()
 
-    def _normalize_target_temp_for_hvac_mode(self):
+    def _normalize_target_temp_for_hvac_mode(self) -> None:
         # Set a target temperature if we don't have any
         # This can happen when we switch from Off to On
         if self._target_temp is None:
             self._target_temp = self._tado_zone_data.current_temp
         elif self._current_tado_hvac_mode == CONST_MODE_COOL:
-            if self._target_temp > self._cool_max_temp:
+            if (
+                self._target_temp is not None
+                and self._cool_max_temp is not None
+                and self._target_temp > self._cool_max_temp
+            ):
                 self._target_temp = self._cool_max_temp
-            elif self._target_temp < self._cool_min_temp:
+            elif (
+                self._target_temp is not None
+                and self._cool_min_temp is not None
```

Comment:
instead of checking that on every `if` would it be possible to check it (and set it) in the beginning like `_target_temp`

Reply:
Nice addition! I need to check if the `_target_temp` is a float. I can do that in the first if and save some code. Thanks!